src/libsyntax/parse/attr.rs

```diff
@@ -18,6 +23,40 @@ const DEFAULT_UNEXPECTED_INNER_ATTR_ERR_MSG: &str = "an inner attribute is not \
                                                      permitted in this context";
 
 impl<'a> Parser<'a> {
+    crate fn parse_arg_attributes(&mut self) -> PResult<'a, Vec<ast::Attribute>> {
+        let mut attrs: Vec<ast::Attribute> = Vec::new();
+        loop {
```

Comment:
For better readability, I would suggest extracting the one-step solution out into its own method i.e. `(&mut self) -> PResult<'a, ast::Attribute>` and then use that in the `loop`. I think that requires less mental state.

Reply:
For better readability I'd suggest to remove this method because the check should be [done in AST validation](https://github.com/rust-lang/rust/pull/60669/files#r283895122) :)

Reply:
`parse_outer_attributes` should be used instead of parsing attributes one-by-one.

Reply:
`parse_outer_attributes` collects commentaries and it isn't possible to doc comment (`///`) [inside an attribute](https://github.com/rust-lang/rust/blob/9ebf47851a357faa4cd97f4b1dc7835f6376e639/src/test/ui/parser/fn-arg-doc-comment.stderr).

Reply:
Doc comments are going to be rejected by the attribute name check in AST validation.
(`/// Text` is a sugar for `#[doc = "Text"]` and attributes named `doc` are not allowed.)

Reply:
I kept `parse_arg_attributes` (for now) because it is easier to collect for `self.sess.param_attr_spans`.

Reply:
Yeah, keeping `parse_arg_attributes` is good if we need to collect spans, but please still use `parse_outer_attributes`.
You can get the spans from its result.